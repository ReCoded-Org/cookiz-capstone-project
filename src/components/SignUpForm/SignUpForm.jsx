import { useRouter } from "next/router";
import { AiOutlineUser } from "react-icons/ai";
import { TbToolsKitchen } from "react-icons/tb";

export default function SignUpLanding() {
    // initializing router
    const router = useRouter();

    // handle redirictin, takes a Path as an argument
    const handleRedirect = (e, PATH) => {
        e.preventDefault();
        router.push(PATH);
    };

    return (
        // root
        <div className='container mx-auto mt-28 flex flex-col p-10'>
            <h1 className='text-center text-5xl font-bold text-primary'>
                Choose Your Account Type
            </h1>
            {/* chose between user or chef container */}
            <div className='flex w-full flex-row flex-wrap justify-center gap-10 p-10'>
                {/* customer */}
                <div
                    className='h-48 w-48 rounded-3xl border-4 border-primary text-primary hover:cursor-pointer hover:bg-primary hover:text-white'
                    onClick={(e) => handleRedirect(e, "/signUpUser")}
                >
                    {/* icon and text */}
                    <div className='m-9 flex flex-col items-center'>
                        <AiOutlineUser size={80} />
                        <p className='m-1 text-xl font-bold '>Customer</p>
                    </div>

                </div>
                {/* chef */}
                <div
                    className='h-48 w-48 rounded-3xl border-4 border-primary text-primary hover:cursor-pointer  hover:bg-primary hover:text-white'
                    onClick={(e) => handleRedirect(e, "/signUpChef")}
                >
                    {/* icon and text */}
                    <div className='m-9 flex flex-col items-center'>
                        <TbToolsKitchen size={80} />
                        <p className='m-1 text-xl font-bold '>Chef</p>
                    </div>
                </div>
            </div>
        </div>
    );
}
