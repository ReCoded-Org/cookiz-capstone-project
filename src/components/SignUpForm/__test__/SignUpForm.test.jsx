import renderer from "react-test-renderer";

import SignUpForm from "../SignUpForm";

// eslint-disable-next-line
it("Create Account Form Component", () => {
    const tree = renderer.create(<SignUpForm />).toJSON();
    // eslint-disable-next-line no-undef
    expect(tree).toMatchSnapshot();
});
