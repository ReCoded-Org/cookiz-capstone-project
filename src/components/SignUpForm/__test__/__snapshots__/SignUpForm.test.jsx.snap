// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Sign up Landing page Component 1`] = `
<div
  className="container mx-auto mt-28 flex flex-col p-10"
>
  <h1
    className="text-center text-5xl font-bold text-primary"
  >
    Choose Your Account Type
  </h1>
  <div
    className="flex w-full flex-row flex-wrap justify-center gap-10 p-10"
  >
    <div
      className="h-48 w-48 rounded-3xl border-4 border-primary text-primary hover:cursor-pointer hover:bg-primary hover:text-white"
      onClick={[Function]}
    >
      <div
        className="m-9 flex flex-col items-center"
      >
        <svg
          fill="currentColor"
          height={80}
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 1024 1024"
          width={80}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
          />
        </svg>
        <p
          className="m-1 text-xl font-bold "
        >
          Customer
        </p>
      </div>
    </div>
    <div
      className="h-48 w-48 rounded-3xl border-4 border-primary text-primary hover:cursor-pointer hover:bg-primary hover:text-white"
      onClick={[Function]}
    >
      <div
        className="m-9 flex flex-col items-center"
      >
        <svg
          fill="none"
          height={80}
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width={80}
          xmlns="http://www.w3.org/2000/svg"
        >
          <desc />
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <path
            d="M4 3h8l-1 9h-6z"
          />
          <path
            d="M7 18h2v3h-2z"
          />
          <path
            d="M20 3v12h-5c-.023 -3.681 .184 -7.406 5 -12z"
          />
          <path
            d="M20 15v6h-1v-3"
          />
          <line
            x1="8"
            x2="8"
            y1="12"
            y2="18"
          />
        </svg>
        <p
          className="m-1 text-xl font-bold "
        >
          Chef
        </p>
      </div>
    </div>
  </div>
</div>
`;
